### PR TITLE
:tada: Sort the blocks when saving as ipynb

### LINKED ISSUE #141: The ordering of the ipynb file should make sense when saving as ipynb
Currently, the order is random, making this feature misleading.

### PR DESCRIPTION
**Pull request by FabienRoger**
Fixes #141

### CODE REVIEW

**Review by MathisFederico — COMMENTED** [review]
We need a test for this feature.
Moreover, we need a test that ensure the conversion from .ipyg to .ipynb is reversible (using .ipynb metadata for edges & non-code/markdown blocks).

**Comment by FabienRoger** [comment]
The current conversion is reversible if there is no markdown blocks.
I agree that metadata should be added so that no information is lost when converting from ipynb and back. But this is not strictly necessary, whereas this current pull request renders the "save as" function usable.

Therefore, I think that adding reversibility should be its own issue/PR.

File: tests/unit/scene/test_to_ipynb_converstion.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    def test_empty_data(self, mocker: MockerFixture):
+        """should return an empty list of cells when the ipyg graph is empty."""
+        ipynb_data = ipyg_to_ipynb({"blocks": [], "edges": []})
+        check.equal("cells" in ipynb_data, True)
+        check.equal(type(ipynb_data["cells"]), list)
+        check.equal(len(ipynb_data["cells"]), 0)
+
+    def test_mnist_example(self, mocker: MockerFixture):
+        """should return the right number of cells in a right order on the mnist example."""
```
Comment (MathisFederico) [inline tests/unit/scene/test_to_ipynb_converstion.py]: I feel like the mnist file is too big for a unit test ...
Maybe take the "simple" .ipynb asset and add a branch in a graph ?

This makes a good integration test though !

**Comment by MathisFederico** [comment]
> The current conversion is reversible if there is no markdown blocks. I agree that metadata should be added so that no information is lost when converting from ipynb and back. But this is not strictly necessary, whereas this current pull request renders the "save as" function usable.
> 
> Therefore, I think that adding reversibility should be its own issue/PR.

Fine by me, but create an issue then !

File: tests/assets/mnist.ipyg [hunk 2]
```diff
@@ -0,0 +1,457 @@
+{
```
Comment (MathisFederico) [inline tests/assets/mnist.ipyg]: Do not duplicate the file if not necessary, you can use the example file for integration tests. (Although not for unit tests, so you were right in your way)

**Review by MathisFederico — CHANGES_REQUESTED** [review]

File: tests/assets/mnist.ipyg (hunk 2 above)
Comment (FabienRoger) [reply tests/assets/mnist.ipyg]: Done!

File: tests/unit/scene/test_to_ipynb_converstion.py (hunk 1 above)
Comment (FabienRoger) [reply tests/unit/scene/test_to_ipynb_converstion.py]: Done!

**Review by MathisFederico — APPROVED** [review]
Nice feature !
Well done !

**Pull request merged**